zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -50,7 +58,7 @@ public void put(final KeyType key, final ValueType value) {
 
   @Override
   public void put(final DbContext context, final KeyType key, final ValueType value) {
-    transactionDb.put(handle, context, key, value);
+    transactionDb.put(handle, context, compositeKey, value);
```

Comment:
The TransactionalColumnFamily doesn't really align well with the ColumnFamily interface. The keys passed into TransactionalColumnFamily are never used (explicitly) so having them is not only unnecessary but possibly misleading. If we pass a different key when calling an operation, the result would be unexpected. Maybe we can change the interface to reflect the new semantics. wdyt?

Reply:
I kind of agree. Tbh I just did the minimal things to keep the changes and the review small. Sorry if this was confusing to you then. Are you fine when I create a follow up issue for refactoring it? I happily do it, but would like to do it separately.

Reply:
Absolutely 👍 it's not urgent, it that can be a follow-up issue. Thanks

Reply:
thanks I created https://github.com/zeebe-io/zeebe/issues/6206